sdk/python/kubeflow/katib/api/katib_client.py

```diff
@@ -0,0 +1,291 @@
+      :param namespace: defaults to current or default namespace.
+      :return: status str
+    """
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+
+    katibexp = self.get_experiment(name, namespace=namespace)
+    last_condition = katibexp.get("status", {}).get("conditions", [])[-1]
+    return last_condition.get("type", "")
+
+  def is_experiment_succeeded(self, name, namespace=None):
+    
+    """Returns true if experiment succeeded; false otherwise.
+    Args:
+      :param name: An experiment name. required
+      :param namespace: defaults to current or default namespace. optional
+      :return: status bool
+    """
+    experiment_status = self.get_experiment_status(name, namespace=namespace)
+    return experiment_status.lower() == "succeeded"
+
+  def list_trials(self, name=None, namespace=None):
+    
+    """
+    Get trials of an experiment
+    :param name: existing experiment name
+    :param namespace: defaults to current or default namespace
+    :return: trials name with status as list
+    """   
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+    
+    thread = self.api_instance.list_namespaced_custom_object(
+        constants.EXPERIMENT_GROUP,
+        constants.EXPERIMENT_VERSION,
+        namespace=namespace,
+        plural=constants.TRIAL_PLURAL,
+        async_req=True)
+    
+    katibtrial = None
+    try:
+      katibtrial = thread.get(constants.APISERVER_TIMEOUT)
+      result = []
+      for i in katibtrial.get("items"):
+          output = {}
+          if i.get("metadata", {}).get("ownerReferences")[0].get("name") == name :
+              output["name"] = i.get("metadata", {}).get("name")
+              output["status"] = i.get("status", {}).get("conditions", [])[-1].get("type")
+              result.append(output)
+    except multiprocessing.TimeoutError:
+      raise RuntimeError("Timeout trying to getkatib experiment.")
+    except client.rest.ApiException as e:
+      raise RuntimeError(
+        "Exception when calling CustomObjectsApi->get_namespaced_custom_object:\
+          %s\n" % e)
+    except Exception as e:
+      raise RuntimeError(
+          "There was a problem to get experiment {0} in namespace {1}. Exception: \
+          {2} ".format(name, namespace, e))
+    return result
+
+  def get_optimal_hyperparmeters(self, name=None, namespace=None):
+    """
+    Get status, currentOptimalTrial with paramaterAssignments
+    :param name: existing experiment name
+    :param namespace: defaults to current or default namespace
+    :return: dict with status, currentOptimalTrial with paramaterAssignments of an experiment
+    """    
+    if namespace is None:
+      namespace = utils.get_default_target_namespace()
+
+    katibexp = self.get_experiment(name, namespace=namespace)
+    result = {}
+    result["status"] = katibexp.get("status", {}).get("conditions", [])[-1].get("type")
```

Comment:
Do we want to return Experiment status here? Since we have `get_experiment_status` API ?

Reply:
I think it is needed because we are able to get optimized hyperparameter even only some of the trials are completed.

Reply:
In that case, maybe users should handle this situation? For example, if `get_optimal_hyperparmeters` return [] run `list_trials` to check Trial statuses.
How experiment status can help user with it?
Experiment status can be running, but some Trials can be finished.

Reply:
ok will remove status from optimized hyperparameter API

Reply:
removed status from optimized hyperparameter API